test/newton1d.jl

```diff
@@ -0,0 +1,41 @@
+
+using IntervalArithmetic, IntervalRootFinding
+using ForwardDiff
+using Base.Test
+
+const D = IntervalRootFinding.derivative
+
+setprecision(Interval, Float64)
+float_pi = @interval(pi)
+
+setprecision(Interval, 10000)
+big_pi = @interval(pi)
+# Using precision "only" 256 leads to overestimation of the true roots for `cos`
+# i.e the Newton method gives more accurate results!
+
+half_pi = big_pi / 2
+three_halves_pi = 3*big_pi/2
+
+@testset "Testing newton1d" begin
+
+    rts =  newton1d(sin, cos, -5..5)
```

Comment:
You should repeat each test where you use AD to calculate the derivative.
You can do this using a loop.

Reply:
The test case fails for f(x) = e^(x^2) - cos(x) as I don't think AD is able to calculate the derivative for that function, should I exclude that from AD?